### DISCUSSION TITLE
Switching Colors based on exec output value

### DISCUSSION THREAD

**Discussion by Dutt-A** [Alexays/Waybar#3454] [Q&A] [answered]
Hello,
I'm trying to make a simple indicator for [rofication](https://github.com/DaveDavenport/Rofication) that indicates the current number of unread notifications.

Essentially, `rofication` gives a list of notifications with dmenu that you have to manually sift through and clear, and `rofication-status` gives you the number of unread notifications.

I'm trying to change the `background-color` of my `custom-rofication` style to be a different color if `rofication-status` returns 0. I know from [here](https://github.com/Alexays/Waybar/wiki/Module:-Custom#module-custom-config-return-type) that I should be able to change my class value to be the output of my exec script and have this work, but I'm not really understanding how to do that.

Here is my current attempt:

_style.css_
```
#custom-rofication.0 {
    color: #2e3440;
    background-color: #81a1c1; /* default bg color */
}   

#custom-rofication {
    color: #2e3440; 
    background-color: #d08770; /* nord orange */
}
```

_config:_
```
"custom/rofication": {
    "format": "{} ", /* this is a unicode character for an eye */
    "class": "{}",
    "max-length": 20,
    "interval": 2,
    "exec": "rofication-status",
    "on-click": "rofication"
},
```

EDIT: Solution with intended behavior based on @RobertMueller2 's response. I will likely release this as a fork to the original rofication.

_rofi_notification_daemon_
```
#!/usr/bin/env python3
import sys
import socket

client = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
client.connect("/tmp/rofi_notification_daemon")
client.sendall(bytes("num",'utf-8'))
val = client.recv(32)
val = val.decode('utf-8')
l = val.split('\n',2)
     
print('{"text": "%s", "class": "%s"}' % (str(l[0]), ("no-notif" if l[0]=='0' else "pending-notif")))

#
# This prints `{"text": "0", "class": no-notif}` if we have no notifications, otherwise
# prints `{"text": "# of notifications", "class": no-notif}`
#
```
_config_
```
"custom/rofication": {
  "format": "{} ",
  "max-length": 20,
  "interval": 1,
  "exec": "rofication-status-waybar",
  "return-type": "json",
  "on-click": "rofication"
} 
```
_style_
```
#custom-rofication.pending-notif {
    color: #2e3440;
    background-color: #d08770; /* nord orange */
}

#custom-rofication {
    color: #2e3440;
    background-color: #81a1c1; /* default bg */
}
```

With my SauceCodeProNerdFont, my waybar now looks like this with 0 and 1 notifications respectively.
![image](https://github.com/user-attachments/assets/3bbba9ed-1149-4825-a690-ff919a0bceca)
![image](https://github.com/user-attachments/assets/f337edcf-0710-4766-a8ed-4c21868babbc)

**Comment by RobertMueller2** [comment, marked as answer]
Well, this should work. I'm not sure what to do with the "eye" character and the notification icon my examples already had, but I think it's close enough. ;) I also don't have rofication, so just using a mock script that reacts to an existing text file, the `[ -e _testfile ]` can be replaced with `rofication > /dev/null 2>&1`

```
{
    "layer": "top", // Waybar at top layer
    "position": "top", // Waybar position (top|bottom|left|right)
    "height" : 50,
    "modules-center": ["custom/rofication"],

    "custom/rofication": {
        "format": "{} ", /* this is a unicode character for an eye */
        "interval": 2,
        "exec": "./rofication-status",
        "on-click": "rofication",
    },
}
```

```
#!/bin/sh

TEXT="";
TOOLTIP="notifications: 0";
CLASS="none"

if [ -e _testfile ];then
  TEXT="";
  TOOLTIP="Notifications: 1";
  CLASS="some"
fi

echo "${TEXT}\n${TOOLTIP}\n${CLASS}"
```

```
#custom-rofication {
    background-color: #dc322f; /*red*/
}

#custom-rofication.none {
    background-color: #268bd2; /* blue */
}
```

![satty-20240720-06:26:59](https://github.com/user-attachments/assets/b5f8877e-a8c2-44ed-93a5-262fd7fbddaf)
![satty-20240720-06:27:16](https://github.com/user-attachments/assets/28322ae1-3c3f-4bf5-9589-6a49c51ac0f4)

If you want to use `"return-type" : "json"`, you can use `printf '{"text": "%s", "tooltip": "%s", "class": "%s"}
' "$TEXT" "$TOOLTIP" "$CLASS"` instead of `echo`.

Another idea, with json and `"hide-empty-text": true` in 0.10.4:
```
#!/bin/sh

if [ -e _testfile ];then
  TEXT="";
  TOOLTIP="Notifications: 1";
  CLASS="some"
  printf '{"text": "%s", "tooltip": "%s", "class": "%s"}\n' "$TEXT" "$TOOLTIP" "$CLASS"
fi

printf '{}'

```

This disables the module altogether, no classes needed.

![satty-20240720-06:28:38](https://github.com/user-attachments/assets/e58d9a6d-dc68-4b93-8fa2-d6807c32fa16)

**Reply by Dutt-A** [reply]
Thanks! I was able to figure out exactly what to do from your code! I will write down what I have.
